### ISSUE TITLE
Support RuboCop 1.0

### ISSUE THREAD

**Issue by teoljungberg** [codeclimate/codeclimate-rubocop#262]
Last week, RuboCop shipped their 1.0 release: https://github.com/rubocop-hq/rubocop/releases/tag/v1.0.0

As this is a big milestone semver-wise, I think it makes sense to support 1.0 as the default RuboCop version for codeclimate. There will be no breaking changes in terms of introducing new cops for RuboCop until 2.0 ships.

More on that in the following blogpost by the maintainer: https://metaredux.com/posts/2020/10/21/rubocop-1-0.html

A good first step would be to add support for 1.0 in a release channel.

**Comment by britneywright** [comment]
Looks like 1.1.0 was released within the hour with some new cops https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md#110-2020-10-29 maybe we should go straight to that.

**Comment by britneywright** [comment]
Looks like there are some gems that have not yet released versions that are compatible with RuboCop >= 1.0.0
- `rubocop-i18n` has an open PR https://github.com/puppetlabs/rubocop-i18n/pull/46/files
- I don't see any activity in `rubocop-sequel`
The rest of the gems seems to support upgrading, at least in terms of bundling.

**Comment by wagenet** [comment]
rubocop-sequel has been updated, still waiting on rubocop-i18n

**Comment by jsdalton** [comment]
Note: https://github.com/puppetlabs/rubocop-i18n has been archived. So I would presume codeclimate would either need to fork and modify or remove that dependency for this service in order to unblock this issue from being resolvable.

Not sure if any codeclimate employees are in this thread, but we're a bit frustrated that the rubocop plugin has fallen so far behind the current branch of Rubocop. This is one of our main use cases as customers.

**Comment by ThomasKoppensteiner** [comment]
@jsdalton a few hours after your post the repo was unarchived. Now @highb takes care of it.

**Comment by imajes** [comment]
@britneywright can we take a look at this? It'd be really swell to get this updated for the 1.x versions :D

**Comment by wagenet** [comment]
Rubocop 1.x is now supported. This can be closed.

**Comment by imajes** [comment]
It would be good to actually add the 1.8.1 engine channel to codeclimate.com and change "rubocop-1.70" to "rubocop-1-7-1" to correctly match the channel name

**Comment by sdrioux** [comment]
^ This.  

This version is listed as a supported channel here: https://github.com/codeclimate/codeclimate-rubocop/branches/all?utf8=%E2%9C%93&query=channel%2Frubocop - but it looks like the latest version is 1.7.0.  Is there something that needs to be done to add the 1.8.1 channel to codeclimate?

```

Channel rubocop-1-8-1 not found for rubocop, available channels: ["stable", "cache-support", "rubocop-0-42", "rubocop-0-46", "rubocop-0-48", "rubocop-0-49", "rubocop-0-50", "rubocop-0-51", "rubocop-0-52", "rubocop-0-54", "rubocop-0-55", "rubocop-0-56", "rubocop-0-57", "rubocop-0-58", "rubocop-0-59", "rubocop-0-60", "rubocop-0-61", "rubocop-0-62", "rubocop-0-63", "rubocop-0-64", "rubocop-0-66", "rubocop-0-65", "rubocop-0-67", "rubocop-0-68", "rubocop-0-69", "rubocop-0-70", "rubocop-0-71", "rubocop-0-72", "rubocop-0-73", "rubocop-0-74", "rubocop-0-75", "rubocop-0-76", "rubocop-0-76-airbnb", "rubocop-0-77", "rubocop-0-78", "rubocop-0-79", "rubocop-0-80", "rubocop-0-81", "rubocop-0-82", "rubocop-0-83", "rubocop-0-84", "rubocop-0-85", "rubocop-0-86", "rubocop-0-87", "rubocop-0-88", "rubocop-0-89", "rubocop-0-90", "rubocop-0-92", "rubocop-1-70"]
--

**Comment by imajes** [comment]
@codeclimate @ale7714 @DanielWright -- thanks for getting the channel names correct, but where did you announce that? It'd be great if you could help by communicating too! :)

**Comment by sdrioux** [comment]
Agreed - looks like 1.8.1 is now working as well, but would be great to be able to see that somewhere.

**Comment by brandonrdn** [comment]
Hello all, I noticed a bunch of branches here for newer rubocop releases (i.e, 1-12-1, 1-18-3, etc) that haven't been "released" yet, as the newest available version for codeclimate is 1-9-1. I'm curious if there is an expected date to release these newer versions, or if there's anywhere to actively track progress on these. Thanks in advance.

**Comment by mileslane** [comment]
I maintain our rubocop version for several different repositories and have yet to find any clear method for determining what rubocop channel will work.  I always check here:  https://docs.codeclimate.com/docs/rubocop and look for the most current channel.  It would be extremely helpful if all the branches listed actually worked.
I am seeing this now:  `WARNING: unknown engine <rubocop:rubocop-1-10-0>` when running `codeclimate engines:install`

**Comment by fede-moya** [comment]
@mileslane that's because the CLI hasn't been updated yet, will try to do that soon.
